rotkehlchen/tests/api/test_skipped_events.py

```diff
@@ -136,25 +136,25 @@ def test_skipped_external_events(rotkehlchen_api_server_with_exchanges, globaldb
     # now let's test the export skipped external events endpoint
     flat_general = KRAKEN_GENERAL_LEDGER_RESPONSE.translate({ord(c): None for c in string.whitespace})  # noqa: E501
     flat_stake = input_ledger.translate({ord(c): None for c in string.whitespace})
-    with tempfile.NamedTemporaryFile(delete=True, suffix='.csv') as tempcsvexport:
-        temppath = Path(tempcsvexport.name)
-        response = requests.put(
-            api_url_for(server, 'historyskippedexternaleventresource'),
-            json={'filepath': str(temppath)},
-        )
-        assert_simple_ok_response(response)
-        with open(temppath, newline='', encoding='utf-8') as csvfile:
-            reader = csv.DictReader(csvfile)
-            count = 0
-            for row in reader:
-                count += 1
-                assert row['location'] == 'kraken'
-                assert row['extra_data'] == '{"location_label":"mockkraken"}'
-                if 'D3' in row['data'] or 'W3' in row['data']:
-                    assert row['data'] in flat_general
-                else:
-                    assert row['data'] in flat_stake
-        assert count == 5
+
+    export_dir = Path(tmpdir_factory.mktemp('test_csv_dir'))
```

Comment:
you don't really need a fixture for the tempdir factory. You can still use the same stdlib I used before just with https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory instead.

But also don't think what you did is wrong so does not need to change.